src/mesido/workflows/io/write_output.py

```diff
@@ -909,6 +898,36 @@ def _name_to_asset(name):
                 *self.energy_system_components.get("heat_exchanger", []),
                 *self.energy_system_components.get("heat_pump", []),
             ]:
+                # Note: when adding new variables to variables_one_hydraulic_system or"
+                # variables_two_hydraulic_system also add quantity and units to the ESDL for the new
+                # variables in the code lower down
+                # These variables exist for all the assets. Variables that only exist for specific
+                # assets are only added later, like Pump_power
+                variables_one_hydraulic_system = ["HeatIn.Q", "Heat_flow"]
+                variables_two_hydraulic_system = [
+                    "Primary.HeatIn.Q",
+                    "Secondary.HeatIn.Q",
+                    "Heat_flow",
+                ]
+
+                # Update/overwrite each asset variable list due to:
+                # - the addition of head loss minimization: head variable and pump power
+                # - only a specific variable required for a specific asset: pump power
+                if self.heat_network_settings["minimize_head_losses"]:
+                    variables_one_hydraulic_system.append("HeatIn.H")
+                    variables_two_hydraulic_system.append("Primary.HeatIn.H")
+                    variables_two_hydraulic_system.append("Secondary.HeatIn.H")
+                    if asset_name in [
+                        *self.energy_system_components.get("heat_source", []),
+                        *self.energy_system_components.get("heat_buffer", []),
+                        *self.energy_system_components.get("ates", []),
```

Comment:
we have pumps for every asset that adds energy to a hydraulic system:
secondary side of heat_exchanger and heat_pump

Reply:
Done